hazelcast/src/test/java/com/hazelcast/jet/core/JobTimeoutTest.java

```diff
@@ -89,11 +89,17 @@ public void when_jobIsResumedAndExceedsTimeout_jobIsCancelled() {
         final JobConfig jobConfig = new JobConfig().setTimeoutMillis(1000L);
         final Job job = hz.getJet().newJob(streamingDag(), jobConfig);
 
-        assertJobStatusEventually(job, RUNNING);
-        job.suspend();
-
-        assertJobStatusEventually(job, SUSPENDED);
-        job.resume();
+        // If the job times out before or during execution of these operations
+        // catch and ignore the error to continue testing. The job should eventually finish with CancellationException
+        // due to timeout
+        try {
+            assertJobStatusEventually(job, RUNNING, 10);
+            job.suspend();
+
+            assertJobStatusEventually(job, SUSPENDED, 10);
+            job.resume();
+        } catch (Throwable ignored) {
```

Comment:
Maybe you should check if the thrown exception is the one you expect

Reply:
Something like this ?
`catch (AssertionError | IllegalStateException ignored) {
        }`

Reply:
rev : a6f6c21faf15762372d360d533daa6fb95b9dbd5

Reply:
AssertionError and IllegalStateException are very generic. Maybe you can assert exception message in the `catch` block?

Reply:
I can not possibly assert all exception messages. There are a lot of combinations. Message from suspend() , message from resume(), assertion message etc.  All we care in the test is the job finishes with CancellationException because of timeout even if suspend() or resume() or assertions don't work